[WIP] Guide for contributing to docs and plugin docs (#2444)
kestra-io/docs: content/docs/docs-contributor-guide.md

This guide is designed to help external contributors get up to speed with the tools, conventions, and components you'll encounter when contributing to Kestra's documentation.

## Contribute to the Kestra Documentation

### ChildCard

The documentation is structured on multiple levels. The top level is an index page such as "Getting Started," "Workflow Components," and "Cloud & Enterprise Edition." This acts as a landing page for all content that falls under those high-level categories. To serve a visitor everything within that topic, we use a `ChildCard` component on the index page. This component is built from the `ChildCard.vue` file within `components.content`.

The index file's markdown looks like this:

```markdown
---
title: Getting Started
---

Follow the [Quickstart Guide](./01.quickstart.md) to install Kestra and start building your first workflows.

::ChildCard
::
```

And the page displays the following with all the sub topics of "Getting Started" listed with their card and icon:

![Getting Started ChildCard](/docs/docs-contributor-guide/child-card.png)

> Important to note when writing a standalone documentation page, the first sentence appears in the ChildCard view to introduce the topic. In the above example for [Quickstart Guide](01.getting-started/01.quickstart.md) this sentence is:

Reviewer: ```suggestion
Important to note when writing a standalone documentation page, the first sentence appears in the ChildCard view to introduce the topic. In the above example for [Quickstart Guide](01.getting-started/01.quickstart.md) this sentence is visible:
```

```markdown
Start Kestra in a Docker container and create your first flow.
```

> Ideally, keep this first sentence as clear and concise as possible so as not to clutter the view on the card.

Reviewer: ```suggestion
Ideally, keep this first sentence as clear and concise as possible to not clutter the view on the card.
```

```java
public class ChatCompletion extends AbstractTask implements RunnableTask<ChatCompletion.Output> {
    @Schema(
        title = "A list of messages comprising the conversation so far.",
        description = "Required if prompt is not set."
    )
    private Property<List<ChatMessage>> messages;

    @Schema(
        title = "The function call(s) the API can use when generating completions."
    )
    private Property<List<PluginChatFunction>> functions;

    @Schema(
        title = "The name of the function OpenAI should generate a call for.",
        description = "Enter a specific function name, or 'auto' to let the model decide. The default is auto."
    )
    private Property<String> functionCall;

    @Schema(
        title = "The prompt(s) to generate completions for. By default, this prompt will be sent as a `user` role.",
        description = "If not provided, make sure to set the `messages` property."
    )
```

> If any titles, descriptions, or examples could be improved or added, create a pull request or an issue on the specific plugin repo, and we will happily review and merge it into the set.

Reviewer: ```suggestion
If any titles, descriptions, or examples could be improved or added, create a pull request or an issue on the specific plugin repository, and we will happily review and merge it into the set.
```

    average price per brand.

    The flow has three tasks:
    1. The first task downloads a JSON file.
    2. The second task filters the data and writes it to a new JSON file.
    3. The third task reads the filtered data, calculates the average price per
    brand using DuckDB, and stores the result as a Kestra output which can be
    previewed and downloaded from the UI.
  tags:
    - Getting Started
    - API
    - Python
    - SQL
  ee: false
  demo: true
  meta_description: This flow represents a data engineering use case. It downloads
    a JSON file, filters the data in Python, and calculates the KPIs in SQL
    using DuckDB.
```

Check out the [full file](https://github.com/kestra-io/blueprints/blob/main/flows/data-engineering-pipeline.yaml) to see the Flow's YAML. 

For the Blueprint to be easily searchable, it is essential to include the appropriate `tags`. A complete list of tags is available on the [Blueprints homepage](/blueprints).

With the proper YAML and extension, the Flow's topology will display interactively on the Blueprint page along with a **Copy source code** button and task icons.

![Blueprint Page](/docs/docs-contributor-guide/blueprint-page.png)

> If you have ideas for new blueprints or existing titles, descriptions, or examples that could be improved, create a pull request or an issue on the Blueprints repo. We will happily review and merge it into the set.

Reviewer: ```suggestion
If you have ideas for new blueprints or existing titles, descriptions, or examples that could be improved, create a pull request or an issue on the [Blueprints repository](http://github.com/kestra-io/blueprints). We will happily review and merge it into the set.
```
